stage0/src/dice.rs

```diff
@@ -0,0 +1,80 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::image_header::Image;
+use core::str::FromStr;
+use dice_crate::{
+    AliasCert, AliasData, AliasOkm, Cdi, CdiL1, DeviceIdOkm, DeviceIdSelfCert,
+    Handoff, SeedBuf, SerialNumber,
+};
+use lpc55_pac::Peripherals;
+use salty::signature::Keypair;
+use sha3::{Digest, Sha3_256};
+use unwrap_lite::UnwrapLite;
+
+fn get_deviceid_keypair(cdi: &Cdi) -> Keypair {
+    let devid_okm = DeviceIdOkm::from_cdi(cdi);
+
+    Keypair::from(devid_okm.as_bytes())
+}
+
+// TODO: get the legit SN from somewhere
+// https://github.com/oxidecomputer/hubris/issues/734
+fn get_serial_number() -> SerialNumber {
+    SerialNumber::from_str("0123456789ab").expect("SerialNumber::from_str")
+}
+
+pub fn run(image: &Image) {
+    let cdi = match Cdi::from_reg() {
+        Some(cdi) => cdi,
+        None => return,
```

Comment:
If the CDI isn't available, this is going to skip turning on the USB SRAM.  That means that any task that attempts to access the DICE artifacts in stage1 will fault when they try to access it.  Probably better to enable the USB SRAM and write out a well-known pattern indicating that DICE was disabled in stage0 so that can reported upstream.

Reply:
This came up elsewhere in the review and I :100: agree. For now the well-known pattern is 0's but I'll do some testing to see how well this works in practice.